### PR TITLE
setup repo & prep v0.1.0

### PR DESCRIPTION
**Pull request by riastrad**
This PR adds all of the code & stands up the important parts of the repo config. Here's a quick summary:

* adds all of the initial working code & tests
* sets up initial package.json for npm (though it won't be released until after we merge this)
* sets up Travis for the repo

**still to do**
- [x] update README.md
- [x] add CODEOWNERS

### CODE REVIEW

**Comment by riastrad** [comment]
@samanpwbb @tristen this is ready for a formal review whenever either of you have a cycle to spare. Once we're 👍 I'll merge & run out a formal release of the new package.

File: index.js
Comment (tristen) [inline index.js:11]: Curious why this isn't the other way around. Wouldn't it be more common to initialize with ` { style, token }` and then call it repeatedly with `coordinates`, and `zoom`?

File: index.js
Comment (tristen) [inline index.js:13]: nit but this could just return? The named variable is clearer though.

File: index.js
Comment (tristen) [inline index.js]: nit but these chains can just be written out as 

```suggestion
      .then(imageIngredients => getAllTiles(imageIngredients, style)); 
```

File: src/get-all-static-tiles.js
Comment (tristen) [inline src/get-all-static-tiles.js]:
```suggestion
  tiles.forEach(tile => {
```

File: src/get-all-static-tiles.js
Comment (tristen) [inline src/get-all-static-tiles.js]:
```suggestion
  return Promise.all(promises).catch(e => {
```

Comment (tristen) [inline src/get-all-static-tiles.js]: `promises` could also just be a `tiles.map`

File: src/get-sibling-tiles-and-offsets.js
Comment (tristen) [inline src/get-sibling-tiles-and-offsets.js]: Would this ever catch?

File: src/get-sibling-tiles-and-offsets.js
Comment (tristen) [inline src/get-sibling-tiles-and-offsets.js]: Would this just be `>` to factor the `maxTiles`?

File: src/stitch-tiles.js
Comment (tristen) [inline src/stitch-tiles.js]: Thoughts on just returning the base64 encoded string or binary? Personally, I feel like consumers should create the png if they need to. quilt can keep pretty lean by not including `pngjs` as a dependency.

File: utils/check-token.js
Comment (tristen) [inline utils/check-token.js]: should `tk` be included?

**Comment by daniel-j-h** [comment]
Hello lovely people :hugs: - I had the use for exactly this a while ago and just used rio back then

https://github.com/mapbox/rasterio/issues/1846#issuecomment-565531023

How's this different and why not just using the rio one-liner?

File: utils/check-token.js
Comment (riastrad) [reply utils/check-token.js]: Not sure home frequently a `tk` token is needed for this use case, but can definitely add one 👍

Comment (riastrad) [reply utils/check-token.js]: added in c8d4937

**Comment by riastrad** [comment]
👋 hello @daniel-j-h, thanks for flagging that slick rasterio one-liner! I had no idea rasterio could deal with slippy maps that easily.

I'm not super familiar with using rasterio, but here's the primary difference I can name off the top of my head: `@mapbox/quilt` is written in Node specifically so that we can use it downstream with some of our internal debugging tools that are also written in Node, and we'd prefer not to mix python & node within these tools.

There may be other good reasons, but I would need to be more familiar with the behavior and uses of `rio calc` before I could say definitively.

File: src/get-all-static-tiles.js
Comment (riastrad) [reply src/get-all-static-tiles.js]: Hmm, I get a linting error from our Mapbox config when I do this 🤔

https://github.com/mapbox/eslint-config-mapbox/blob/master/base.js#L15

File: src/get-sibling-tiles-and-offsets.js
Comment (riastrad) [reply src/get-sibling-tiles-and-offsets.js]: Yep, good catch.

File: src/get-all-static-tiles.js
Comment (tristen) [reply src/get-all-static-tiles.js]: Oh! I guess we require parens. Let's just close these suggestions 🙈

File: src/get-all-static-tiles.js
Comment (riastrad) [reply src/get-all-static-tiles.js:17]: `tiles.map` is a great suggestion! 🙌

File: src/get-sibling-tiles-and-offsets.js
Comment (riastrad) [reply src/get-sibling-tiles-and-offsets.js]: 🤔 I don't think so, actually. I'll remove it and see what happens.

File: src/get-all-static-tiles.js
Comment (riastrad) [reply src/get-all-static-tiles.js]: `tiles.map` is a great suggestion! 🙌

File: index.js
Comment (riastrad) [reply index.js:11]: Currently I'm just focused on keeping this as it's currently being used downstream, which is why the pattern is structured this way. But I'm not opposed to inverting the `make/get` relationship, as it does make more sense the way you've outlined.

File: index.js
Comment (riastrad) [reply index.js:13]: Yeah, I purposefully opted for clarity over concision here 👍

File: index.js
Comment (riastrad) [reply index.js]: Great point, changed in 53b2f84 👍

File: src/stitch-tiles.js
Comment (riastrad) [reply src/stitch-tiles.js]: Ooh, I like where your head's at. I _would_ like to keep quilt as light as possible, so I can definitely get on board with pushing responsibility for decoding to downstream consumers. 👍

Comment (riastrad) [reply src/stitch-tiles.js]: Switching to returning just the raw buffer in https://github.com/mapbox/quilt/pull/1/commits/9abde5f45017c0ac14c494e351ad572794241092

File: src/get-sibling-tiles-and-offsets.js
Comment (riastrad) [reply src/get-sibling-tiles-and-offsets.js]: Hmmm, even though tests are passing in my latest commit, I'm seeing some weird false-positive behavior for the get-sibling-tiles-and-offsets tests now: 

Screen Shot 2020-04-29 at 12 03 17 PM
[link](https://travis-ci.com/github/mapbox/quilt/jobs/325085201)

Comment (riastrad) [reply src/get-sibling-tiles-and-offsets.js]: I'll need to dig into this a bit more to understand why this change is causing that behavior, but I suspect it's primarily to do with how the test is written since the change lines did not modify how `zoom` values are handled at all.

Comment (riastrad) [reply src/get-sibling-tiles-and-offsets.js]: Ah, yep - bad fixture. Fixed in 4d7622b

**Comment by riastrad** [comment]
@tristen @samanpwbb this is ready for a final review.

File: index.js
Comment (tristen) [reply index.js:11]: Obvs your call here but is this something you would like to do instead? It's a pretty breaking change if it's introduced later on.

Comment (riastrad) [reply index.js:11]: I did a quick review of the downstream implementations and I think shifting this structure would cause us some non-trivial refactoring for this use case. My gut reaction is to defer as much downstream refactoring as possible, and accept this pattern as is for now.

If we decide to use this library in more places in the future (and find ourselves frustrated by the pattern), then the breaking change would be more worth the effort.

**Review by tristen — APPROVED** [review]
🧶 🖼️

**Pull request merged**
